src/libs/utils.js

```diff
@@ -368,3 +368,13 @@ export const truncateInteger = integer => {
 
   return `${Math.floor(integer / 1000)}k`
 }
+
+// can this be done with usePollingEffect?
```

Comment:
@kyuksel I'd specifically like eyes on this, if possible! I feel like there should probably be an existing way to do this in the codebase, but couldn't figure it out. `usePollingEffect` seemed to be mostly used for infinite polling, or something of that nature.

Reply:
It makes sense to avoid `usePollingEffect` for this so that it can be used outside of a React component.

Reply:
We could, however, probably use this in `usePollingEffect` to simplify that code. I can make that refactor.

Reply:
I agree that refactoring `poll()` so it can be used from within `usePollingEffect()` would be great.

> Not quite out of the box, it looks like - if I am understanding correctly - usePollingEffect also has a variable that says whether the function calls should start after the delay, or before the delay

[Thinking out loudly] I wonder if something like

```javascript
const poll = async (pollFn, pollTime, pollUntil, leading, signal) => {
  const initialResult = leading ? await pollFn() : undefined
  
  if (!_.isUndefined(initialResult) && pollUntil(initialResult)) {
    return initialResult
  }
  
  let result = initialResult
  while (!signal.aborted && !pollUntil(result)) {
    await delay(pollTime)
    result = !signal.aborted && await pollFn()
  }
  
  return result
}
```

could be used by both `usePollingEffect()` and from within `dataBrowser-utils.js`. Note that I haven't tested it.

Either way, the function could probably use [JSDoc](https://en.wikipedia.org/wiki/JSDoc)'ing.